chore(docs): Replace the `dev` docs version with latest `nightly` release docs build

critesjosh opened the pull request:
This PR adds nightly docs releases.

This adds a workflow that will run at 4am UTC every day. The workflow checks that the nightly release tag exists and builds the Aztec and barretenberg docs at that release commit, then commits the changes back to the `next` branch.

Note: this will mean that changes in `docs/docs` will no longer be preview-able in the netlify docs builds.

critesjosh commented on .github/workflows/nightly-docs-release.yml:
```diff
@@ -0,0 +1,157 @@
+        working-directory: ./barretenberg/docs
+        run: |
+          # Set the commit tag for version macros
+          export COMMIT_TAG=${{ env.NIGHTLY_TAG }}
```
@signorecello this COMMIT_TAG isn't needed for barretenberg docs right? This bit was copied from the release-please.json for `master`, but it looks like its not being used.

alejoamiras commented on barretenberg/docs/docusaurus.config.ts:
```diff
@@ -82,11 +82,11 @@ const config: Config = {
+          includeCurrentVersion: false,
+          // There should be 2 versions, nightly and stable
+          // The stable version is second in the list
+          lastVersion: versions[1],
```
Would it make sense to check if we are on `env === dev` so we can have `dev` on local?

critesjosh replied:
yep, updated

alejoamiras commented on .github/workflows/nightly-docs-release.yml:
```diff
@@ -0,0 +1,157 @@
+      should-run: ${{ steps.get_tag.outputs.should_run }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
```
Why are we using `@11bd71901bbe5b1630ceea73d27597364c9af683` 👀 ? Weird checked out version.

critesjosh replied:
i copied this from the release-please.json. https://github.com/AztecProtocol/aztec-packages/blob/b30fe477b02ffe4e54c49835d0f77fd782df246a/.github/workflows/release-please.yml#L145

alejoamiras reviewed (approved):
I'm not seeing any obvious mistakes. I've also tested the scripts locally and they seem to work :) . My only thing would be to also have `dev` on local so we can iterate faster. Unless I miss-understood something of the PR and we are still having that.

sklppy88 reviewed (approved):
Looks good to me, probably unrelated to your PR but it's going to 3.0.0 nightly which is a bit weird to me

critesjosh committed:
chore(docs): Replace the `dev` docs version with latest `nightly` release docs build

This PR adds nightly docs releases.

This adds a workflow that will run at 4am UTC every day. The workflow checks that the nightly release tag exists and builds the Aztec and barretenberg docs at that release commit, then commits the changes back to the `next` branch.

Note: this will mean that changes in `docs/docs` will no longer be preview-able in the netlify docs builds.

Co-authored-by: Josh Crites <jc@joshcrites.com>
